source/extensions/filters/network/dubbo_proxy/decoder.cc

```diff
@@ -64,7 +64,7 @@ DecoderStateMachine::DecoderStatus DecoderStateMachine::handleState(Buffer::Inst
   case ProtocolState::OnDecodeStreamData:
     return onDecodeStreamData(buffer);
   default:
-    NOT_REACHED_GCOVR_EXCL_LINE;
+    PANIC("unexpected");
```

Comment:
What's the rubric for `PANIC` vs `IS_ENVOY_BUG`?

Reply:
details in linked issue but tl;dr PANIC if there' a return and it won't compile today, IS_ENVOY_BUG for void, where we can not crash the server.
Ideally we replace most of the PANICs with sensible default returns - there's that follow-up bug for that.